pkg/prometheus/operator.go

```diff
@@ -1592,6 +1592,13 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 		return errors.Wrap(err, "loading additional alert manager configs from Secret failed")
 	}
 
+	// Validate Prometheus Config Inputs at Prometheus CRD level
+	err = c.configGenerator.ValidateConfigInputs(p)
```

Comment:
I'm not sure we need to create another receiver method for `*ConfigGenerator`. IMHO the validation should be done by `GenerateConfig()` directly so the caller doesn't have to worry about validating th configuration before generating.

Reply:
Did you mean to call the ValidateInput() func from inside GenerateConfig() method?

Reply:
yes and we don't need to make `ValidateConfigInputs()` public (e.g. `validateConfigInputs()` instead).

Reply:
thank you made it public since I was trying to call from operator.go. I am updating with changes